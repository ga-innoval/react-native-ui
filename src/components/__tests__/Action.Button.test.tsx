import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { ActionButton } from '../Action.Button';
const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};

describe('<ActionButton loading={false} />', () => {
  const mockFn = jest.fn();
  let component: any = null;

  beforeEach(() => {
    component = render(
      <ActionButton
        style={{ marginTop: 10 }}
        text="Press me"
        loading={false}
        onPress={mockFn}
      />
    );
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call function on press', () => {
    const button = screen.getByText(/press me/i);
    fireEvent.press(button, eventData);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});

describe('<ActionButton loading={true} />', () => {
  const mockFn = jest.fn();
  let component: any = null;

  beforeEach(() => {
    component = render(
      <ActionButton
        style={{ marginTop: 10 }}
        text="Press me"
        loading={true}
        onPress={mockFn}
      />
    );
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("shouldn't call function on press", () => {
    const button = screen.getByAccessibilityHint('button');
    fireEvent.press(button, eventData);
    expect(mockFn).toHaveBeenCalledTimes(0);
  });
});
