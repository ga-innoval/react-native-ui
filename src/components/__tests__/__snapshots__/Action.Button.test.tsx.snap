// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton loading={false} /> renders correctly 1`] = `
<View
  accessibilityHint="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 15,
      },
      {
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#7A7878",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Press me
  </Text>
  <Text />
</View>
`;

exports[`<ActionButton loading={true} /> renders correctly 1`] = `
<View
  accessibilityHint="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 15,
      },
      {
        "marginTop": 10,
      },
      false,
    ]
  }
>
  <ActivityIndicator
    color="#232222"
    size="small"
  />
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#232222",
          "fontSize": 22,
        },
        undefined,
        {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ó°…‚
  </Text>
</View>
`;
